### PR TITLE
Blog App: Processing Data in Models

### PR DESCRIPTION
**Pull request by gtekle**
### In this pull request, I have:
  - Added has_many associations to Models
  - Added custom methods to Models
  - Added pry-rails gem
  - Created MIT.md license file with updated content

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
+
```
Comment (MahmoudBakr23) [inline app/models/user.rb]: - [ ] Kindly specify the `foreign_key` for the User model's associations here for the application to know that this specific model is what you're calling the `author`.
Ex:
`has_many :posts, foreign_key: 'author_id'`

File: app/models/post.rb [hunk 2]
```diff
@@ -1,3 +1,14 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
```
Comment (MahmoudBakr23) [inline app/models/post.rb:5]: **OPTIONAL**
- It'd be nice if you created a custom method using the `after_save` rails callback to call the `udpate_user_posts_counter` method every time the model gets updated!

Ex:
`after_save :udpate_user_posts_counter`

> Please consider this suggestion in the other models

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @gtekle 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Linters are correct
- Nice job setting the models' association so far
- Great job using the `class_name` to identify the `User` model in the associated models
- Well done limiting the queries returned using custom methods in your models

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] Please fix the MIT link for a completely professional README file

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gtekle** [comment]
Hi @MahmoudBakr23 
Thank you very much for the feedback. I will address all of them.
:pray:

File: app/models/post.rb (hunk 2 above)
Comment (gtekle) [reply app/models/post.rb:5]: Thanks @MahmoudBakr23 , for introducing me to Active Record callbacks! :pray:

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @gtekle,** 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## **Invalid Code Review Request**

You have submitted a project, that _**you have submitted a project that does not contain any details in the PR, and as the general requirement says see highlighted with red**_


Capture web_2-5-2022_95245_gitlab com

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by gtekle** [comment]
oops...! I am not sure how I skipped it. That was a good catch thanks :+1:

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @gtekle, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gtekle** [comment]
Thanks @Mihndim2020 :pray:

**Pull request merged**